src/main/kotlin/uk/gov/dluhc/printapi/service/CertificateRemovalDateResolver.kt

```diff
@@ -0,0 +1,58 @@
+package uk.gov.dluhc.printapi.service
+
+import org.springframework.stereotype.Component
+import uk.gov.dluhc.printapi.client.BankHolidayDataClient
+import uk.gov.dluhc.printapi.client.BankHolidayDivision
+import uk.gov.dluhc.printapi.config.DataRetentionConfiguration
+import java.time.DayOfWeek.SATURDAY
+import java.time.DayOfWeek.SUNDAY
+import java.time.LocalDate
+import java.time.temporal.ChronoUnit
+
+/**
+ * Responsible for determining the removal date for a Certificate's data. The removal date varies, depending on the
+ * category of data and the type of Certificate concerned.
+ */
+@Component
+class CertificateRemovalDateResolver(
+    private val dataRetentionConfig: DataRetentionConfiguration,
+    private val bankHolidayDataClient: BankHolidayDataClient
+) {
+
+    /**
+     * Calculates the date that the delivery info for a Certificate's print requests should be removed. The retention
+     * period is currently 28 (configurable) working days from the certificate's "issue" date, which means weekends and
+     * bank holidays should be excluded when making this calculation. If the issue date was more than 28 working days
+     * ago, the removal date will be in the past.
+     *
+     * @param issueDate The date the Certificate was issued.
+     * @param gssCode The Certificate's GSS code which is used to determine if there are any upcoming bank holidays for
+     * the UK nation concerned.
+     * @return A [LocalDate] representing when a Certificate's delivery info should be removed
+     */
+    fun getCertificateDeliveryInfoRemovalDate(issueDate: LocalDate, gssCode: String): LocalDate =
+        with(getTotalDaysForWorkingDays(issueDate, dataRetentionConfig.certificateDeliveryInfo.toDays(), gssCode)) {
+            issueDate.plusDays(this.toLong())
+        }
+
+    private fun getTotalDaysForWorkingDays(issueDate: LocalDate, requiredWorkingDays: Long, gssCode: String): Int {
```

Comment:
Could this return a `Long` instead of an `Int`? (This removes the  need of `toLong()` on line 35)

Reply:
A long would be inappropriate.